hs_app_timeseries/models.py

```diff
@@ -950,211 +1106,462 @@ def delete_all_elements(self):
-         Django.
-        :param user: current user (must have edit permission on resource) who is updating the
-        sqlite file to sync metadata changes in django database.
-        :return:
-        """
-        log = logging.getLogger()
-        if not self.resource.has_csv_file:
-            msg = "Resource needs to have a CSV file before a blank SQLite file can be updated"
-            log.exception(msg)
-            raise Exception(msg)
+        cur.execute("DELETE FROM Datasets")
+        con.commit()
+        insert_sql = "INSERT INTO Datasets (DatasetID, DatasetUUID, DatasetTypeCV, " \
+                     "DatasetCode, DatasetTitle, DatasetAbstract) VALUES(?,?,?,?,?,?)"
 
-        # update resource specific metadata element series ids with generated GUID
-        self._update_metadata_element_series_ids_with_guids()
+        if isinstance(self, TimeSeriesMetaData):
+            ds_title = self.title.value
+            ds_abstract = self.description.abstract
+            ds_code = self.resource.short_id
+        else:
+            # self must be TimeSeriesFileMetaData
+            ds_title = self.logical_file.dataset_name
+            ds_abstract = self.abstract
+            ds_code = self.logical_file.resource.short_id
 
-        blank_sqlite_file = utils.get_resource_files_by_extension(self.resource, ".sqlite")[0]
-        csv_file = utils.get_resource_files_by_extension(self.resource, ".csv")[0]
+        cur.execute(insert_sql, (1, uuid4().hex, 'Multi-time series', ds_code, ds_title,
+                                 ds_abstract), )
 
-        # retrieve the csv file from iRODS and save it to temp directory
-        temp_csv_file = utils.get_file_from_irods(csv_file)
-        try:
-            con = sqlite3.connect(temp_sqlite_file)
-            with con:
-                # get the records in python dictionary format
-                con.row_factory = sqlite3.Row
-                cur = con.cursor()
-                # insert records to SamplingFeatures table
-                self._update_samplingfeatures_table_insert(con, cur)
+    def update_datatsetsresults_table_insert(self, con, cur):
+        # insert record to DatasetsResults table - first delete any existing records
+        # used for updating a sqlite file that is blank (case of CSV upload)
 
-                # insert record to SpatialReferences table
-                self._update_spatialreferences_table_insert(con, cur)
+        cur.execute("DELETE FROM DatasetsResults")
+        con.commit()
+        insert_sql = "INSERT INTO DatasetsResults (BridgeID, DatasetID, " \
+                     "ResultID) VALUES(?,?,?)"
 
-                # insert record to Sites table
-                self._update_sites_table_insert(con, cur)
+        cur.execute("SELECT ResultID FROM Results")
+        results = cur.fetchall()
+        for index, result in enumerate(results):
+            bridge_id = index + 1
+            cur.execute(insert_sql, (bridge_id, 1, result['ResultID']), )
```

Comment:
Looks like you insert a constant ```1``` as DatasetID into DatasetsResults? Is that correct?

Reply:
Yeah it seems it is always going to be 1. However, I will change it to read the value of the DatasetID from DataSet table and use that instead.

Reply:
That will definitely better, since using a constant is error-prone over the long run.